### PR TITLE
Sandbox: Use toString to create observe and resize script string

### PR DESCRIPTION
**Pull request by walbo**
## What?
Use `toString` on the `observeAndResizeJS` function to create string used in the iframe.

## Why?
- The function will be minimized in production.
- Eslint will check the function.
- Makes is possible to use TypeScript in the function when the component will be converted.

## How?
Used `toString` instead of hardcoding the function as a string.

## Testing Instructions
Test the editor where the `Sandbox` component is used. Ex the `Custom HTML` block in preview mode.

### CODE REVIEW

**Review by mirka — APPROVED** [review]
Interesting! The reasoning makes sense and it seems like this would be beneficial in multiple ways.

From what I could gather, I don't _think_ this would be problematic (security, reliability), but I would feel better if we got one more reviewer's approval as well.

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for refining this @walbo 👍 

The benefits sound promising, and the changes look good to me. I didn't encounter any issues while testing either.

Further to @mirka's comment, maybe it would be prudent to get a final sign-off by one of our security experts.

**Comment by walbo** [comment]
> it would be prudent to get a final sign-off by one of our security experts.

I don't think there shoud be any security conserns, but having a final sign-off doesn't hurt 😄 

Don't know who to request a review from to get that. @aaronrobertshaw Can you add someone from security team as a reviewer?

Would also be good to get someone from the mobile team to make sure it works as expected in the mobile app as well.

**Review by samueljseay — APPROVED** [review]
My 2 cents from a security perspective is that I don't see an additional security threat being introduced here. This function is not modifiable as user input and there is already a way to pass your own function as a prop, a more immediate danger would be users of this component not being aware of the security implications of that.

To abuse this code would require introducing some kind of other XSS vulnerability to the site and at that point there would be easier ways to perform an attack.

**Comment by dcalhoun** [comment]
I opened https://github.com/wordpress-mobile/gutenberg-mobile/pull/5083 to allow a few more mobile-specific CI tests to run against these changes. I hope to spend time testing the changes in the mobile editor within the next couple of days.

File: packages/components/src/sandbox/index.native.js
Comment (dcalhoun) [inline packages/components/src/sandbox/index.native.js:25]: It appears that Hermes requires a [specific directive to return the original source code](https://github.com/facebook/hermes/issues/114#issuecomment-887106990) from `Function.prototype.toString`. Without this, the script failed to execute, resulting in unexpected UI. This only occurs on Android because iOS does not currently leverage Hermes. 

<details><summary>Unexpected UI Screenshots</summary>

| Before | After |
| - | - |
| Twitter embed before is not cut off | Twitter embed after is cut off |
| YouTube embed before contains no gap | YouTube embed after contains large gap |

</details>

Adding the special `'show source'` directive appears to resolve these issues. 

```suggestion
const observeAndResizeJS = function () {
	// Hermes requires a special directive to preserve the original source code
	// when using `Function.prototype.toString()` below.
	// https://github.com/facebook/hermes/issues/114#issuecomment-887106990
	'show source';
```

**Review by dcalhoun — COMMENTED** [review]
The additional mobile CI tasks passed in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5083. 🎉 Unfortunately, I discovered a regression on Android that I have outlined further in a separate comment. 

If we address the Android regression, I do not see a reason why these changes cannot be made to the native mobile file. However, like other commenters, I would defer to others more experienced in security matters to comment on any possible vulnerabilities in these changes. 

Thanks for the ping!

File: packages/components/src/sandbox/index.native.js
Comment (derekblank) [reply packages/components/src/sandbox/index.native.js:25]: Interesting findings, @dcalhoun. I will make a note to check this issue again on iOS and Android as Gutenberg Mobile upgrades React Native (and Hermes) to 0.69.x and beyond.

Comment (walbo) [reply packages/components/src/sandbox/index.native.js:25]: Thanks for thorough testing @dcalhoun! Commited your fix into the branch.

**Comment by ciampo** [comment]
Hey all 👋  Is this PR ready to be merged?

Since some time has passed, should we rebase it before doing so?

**Comment by walbo** [comment]
> Hey all 👋 Is this PR ready to be merged?
> 
> Since some time has passed, should we rebase it before doing so?

I have rebased the PR. I belive it's ready to be merged.

**Comment by ciampo** [comment]
Makes sense to me — let's merge once CI is ✅ !

**Pull request merged**
